src/Core/Grid/Query/MerchandiseReturnProductsQueryBuilder.php

```diff
@@ -0,0 +1,178 @@
+        string $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        int $contextLanguageId
+    ) {
+        parent::__construct($connection, $dbPrefix);
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->contextLanguageId = $contextLanguageId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getMerchandiseReturnProductsQueryBuilder($searchCriteria);
+        $qb->select('
+            ord.id_order_return,
+            ord.id_order_detail,
+            ord.product_quantity,
+            od.product_name,
+            od.product_reference,
+            od.id_customization,
+            o.id_cart'
+            )
+            ->groupBy('ord.id_order_detail');
+
+        $this->searchCriteriaApplicator
+            ->applySorting($searchCriteria, $qb)
+            ->applyPagination($searchCriteria, $qb);
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        return $this->getMerchandiseReturnProductsQueryBuilder($searchCriteria)
+            ->select('COUNT(ord.id_order_detail)');
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    private function getMerchandiseReturnProductsQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'order_return_detail', 'ord')
+            ->leftJoin(
+                'ord',
+                $this->dbPrefix . 'order_detail',
+                'od',
+                'ord.id_order_detail = od.id_order_detail'
+            )
+            ->leftJoin(
+                'od',
+                $this->dbPrefix . 'orders',
+                'o',
+                'od.id_order = o.id_order'
+            )
+            ->setParameter('context_language_id', $this->contextLanguageId);
+
+        $this->applyFilters($searchCriteria->getFilters(), $queryBuilder);
+
+        return $queryBuilder;
+    }
+
+    /**
+     * Apply filters to merchandise returns query builder.
+     *
+     * @param array $filters
+     * @param QueryBuilder $qb
+     */
+    private function applyFilters(array $filters, QueryBuilder $qb)
+    {
+        $allowedFilters = [
+            'product_reference',
+            'product_name',
+            'quantity',
+            'customization_name',
+            'customization_value',
+            'merchandise_return_id',
+        ];
+
+        foreach ($filters as $filterName => $filterValue) {
+            if (!in_array($filterName, $allowedFilters)) {
+                continue;
+            }
+
+            if ($filterName === 'merchandise_return_id') {
+                $qb->andWhere('ord.`id_order_return` LIKE :' . $filterName);
```

Comment:
For a research by ID it feels weird to use a `LIKE` 🤔  I would use a strict equality